src/main/java/de/zalando/aruha/nakadi/controller/SubscriptionController.java

```diff
@@ -55,52 +55,70 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
+        if (!featureToggleService.isFeatureEnabled(FeatureToggleService.Feature.HIGH_LEVEL_API)) {
             return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
         }
         if (errors.hasErrors()) {
-            return create(new ValidationProblem(errors), nativeWebRequest);
+            return Responses.create(new ValidationProblem(errors), request);
         }
 
         try {
-            // check that event types exist
-            final List<String> noneExistingEventTypes = newArrayList();
-            for (final String etName : subscriptionBase.getEventTypes()) {
-                try {
-                    eventTypeRepository.findByName(etName);
-                } catch (NoSuchEventTypeException e) {
-                    noneExistingEventTypes.add(etName);
-                }
-            }
-            if (!noneExistingEventTypes.isEmpty()) {
-                final String errorMessage = "Failed to create subscription, event type(s) not found: '" +
-                        StringUtils.join(noneExistingEventTypes, "','") + "'";
-                LOG.debug(errorMessage);
-                return create(UNPROCESSABLE_ENTITY, errorMessage, nativeWebRequest);
-            }
-
-            // generate subscription id and try to create subscription in DB
-            final Subscription subscription = subscriptionRepository.createSubscription(subscriptionBase);
-            return new ResponseEntity<>(subscription, HttpStatus.CREATED);
-
+            return createSubscription(subscriptionBase, request);
         } catch (final DuplicatedSubscriptionException e) {
             try {
-                // if the subscription with such parameters already exists - return it instead of creating a new one
-                final Subscription existingSubscription = subscriptionRepository.getSubscription(
-                        subscriptionBase.getOwningApplication(), subscriptionBase.getEventTypes(),
-                        subscriptionBase.getConsumerGroup());
-                return new ResponseEntity<>(existingSubscription, HttpStatus.OK);
-
+                return new ResponseEntity<>(getExistingSubscription(subscriptionBase), HttpStatus.OK);
             } catch (final NakadiException ex) {
                 LOG.error("Error occurred during fetching existing subscription", ex);
-                return create(INTERNAL_SERVER_ERROR, ex.getProblemMessage(), nativeWebRequest);
+                return Responses.create(Response.Status.INTERNAL_SERVER_ERROR, ex.getProblemMessage(), request);
             }
         } catch (final InternalNakadiException e) {
             LOG.error("Error occurred during subscription creation", e);
-            return create(e.asProblem(), nativeWebRequest);
+            return Responses.create(e.asProblem(), request);
         }
     }
 
+    private ResponseEntity<?> createSubscription(@Valid @RequestBody final SubscriptionBase subscriptionBase, final NativeWebRequest request)
```

Comment:
As for me this methods should be in SubscribtionService to have business logic outside controllers.